java/system-test/system-test-suite/src/test/java/sleeper/systemtest/suite/QueryIT.java

```diff
@@ -0,0 +1,155 @@
+            assertThat(sleeper.query().direct()
+                    .run("key",
+                            "row-0000000000000000010", "row-0000000000000000030",
+                            "row-0000000000000000020", "row-0000000000000000040"))
+                    .containsExactlyElementsOf(sleeper.generateNumberedRecords(LongStream.range(10, 40)));
+        }
+
+        @Test
+        void shouldRunQueryWithTwoRangesThatDoNotOverlap() {
+            // Given
+            sleeper.ingest().direct(tempDir).numberedRecords(LongStream.range(0, 100));
+
+            // When/Then
+            assertThat(sleeper.query().direct()
+                    .run("key",
+                            "row-0000000000000000010", "row-0000000000000000020",
+                            "row-0000000000000000030", "row-0000000000000000040"))
+                    .containsExactlyElementsOf(sleeper.generateNumberedRecords(LongStream.concat(
+                            LongStream.range(10, 20), LongStream.range(30, 40))));
+        }
+    }
+
+    @Nested
+    @DisplayName("SQS Query")
+    class SQSQuery {
+        @Test
+        void shouldRunQueryForAllRecords() throws InterruptedException {
+            // Given
+            sleeper.ingest().direct(tempDir).numberedRecords(LongStream.range(0, 100));
+
+            // When/Then
+            assertThat(sleeper.query().byQueue()
+                    .allRecordsInTable()
+                    .waitForQuery().results())
+                    .containsExactlyElementsOf(sleeper.generateNumberedRecords(LongStream.range(0, 100)));
+        }
+
+        @Test
+        void shouldRunQueryWithOneRange() throws InterruptedException {
+            // Given
+            sleeper.ingest().direct(tempDir).numberedRecords(LongStream.range(0, 100));
+
+            // When/Then
+            assertThat(sleeper.query().byQueue()
+                    .run("key", "row-0000000000000000010", "row-0000000000000000020")
+                    .waitForQuery().results())
+                    .containsExactlyElementsOf(sleeper.generateNumberedRecords(LongStream.range(10, 20)));
+        }
+
+        @Test
+        void shouldRunQueryWithTwoRangesThatOverlap() throws InterruptedException {
+            // Given
+            sleeper.ingest().direct(tempDir).numberedRecords(LongStream.range(0, 100));
+
+            // When/Then
+            assertThat(sleeper.query().byQueue()
+                    .run("key",
+                            "row-0000000000000000010", "row-0000000000000000030",
+                            "row-0000000000000000020", "row-0000000000000000040")
+                    .waitForQuery().results())
+                    .containsExactlyElementsOf(sleeper.generateNumberedRecords(LongStream.range(10, 40)));
+        }
+
+        @Test
+        void shouldRunQueryWithTwoRangesThatDoNotOverlap() throws InterruptedException {
+            // Given
+            sleeper.ingest().direct(tempDir).numberedRecords(LongStream.range(0, 100));
```

Comment:
It might also be worth controlling the format of the data we generate a bit more so that you can tell the data is correct from looking at the test. Maybe a separate step like `sleeper.formatNumberedRecords(prefixStringRowKeysAndPadDigits("row-", 3))`.